extensions/assets-manager/AssetsManagerEx.cpp

```diff
@@ -494,7 +530,12 @@ void AssetsManagerEx::downloadManifest()
     if (_updateState != State::PREDOWNLOAD_MANIFEST)
         return;
 
-    std::string manifestUrl = _localManifest->getManifestFileUrl();
+    std::string manifestUrl;
+    if (_remoteManifest->isVersionLoaded()) {
+        manifestUrl = _remoteManifest->getManifestFileUrl();
+    } else {
+        manifestUrl = _localManifest->getManifestFileUrl();
+    }
```

Comment:
Um, it's hard to say if this is a bug. It is rather about how it should be.

If we have downloaded the new version file, should we use the one in the new version file or the one already exists locally?